modules/smartadserverBidAdapter.js

```diff
@@ -1,60 +1,93 @@
-        bidObject.currency = adUnit.currency;
-        bidObject.ad = adUnit.ad;
-        bidObject.width = adUnit.width;
-        bidObject.height = adUnit.height;
-        bidObject.dealId = adUnit.dealId;
-        bidmanager.addBidResponse(bid.placementCode, bidObject);
-      } else {
-        utils.logMessage(`[SmartAdServer] no bid response for placementCode ${bid.placementCode}`);
-        bidObject = bidfactory.createBid(2);
-        bidObject.bidderCode = 'smartadserver';
-        bidmanager.addBidResponse(bid.placementCode, bidObject);
-      }
-    };
-    return callbackId;
-  };
+    // if your bidder supports multiple currencies, use config.getConfig(currency)
+    // to find which one the ad server needs
 
-  return {
-    callBids: function(params) {
-      for (var i = 0; i < params.bids.length; i++) {
-        var bid = params.bids[i];
-        var adCall = url.parse(bid.params.domain);
-        adCall.pathname = '/prebid';
-        adCall.search = {
-          'pbjscbk': '$$PREBID_GLOBAL$$.' + generateCallback(bid),
-          'siteid': bid.params.siteId,
-          'pgid': bid.params.pageId,
-          'fmtid': bid.params.formatId,
-          'ccy': bid.params.currency || 'USD',
-          'bidfloor': bid.params.bidfloor || 0.0,
-          'tgt': encodeURIComponent(bid.params.target || ''),
-          'tag': bid.placementCode,
-          'sizes': bid.sizes.map(size => size[0] + 'x' + size[1]).join(','),
-          'async': 1
-        };
-        adloader.loadScript(url.format(adCall));
-      }
+    // pull requested transaction ID from bidderRequest.bids[].transactionId
+    if (!Array.isArray(validBidRequests)) {
+      validBidRequests = [validBidRequests];
     }
-  };
-};
-
-adaptermanager.registerBidAdapter(new SmartAdServer(), 'smartadserver');
-
-module.exports = SmartAdServer;
+    var bid = validBidRequests[0];
+    const payload = {
+      siteid: bid.params.siteId,
+      pageid: bid.params.pageId,
+      formatid: bid.params.formatId,
+      currencyCode: config.getConfig('currency'),
+      bidfloor: bid.params.bidfloor || 0.0,
+      targeting: bid.params.target && bid.params.target != '' ? bid.params.target : undefined,
+      tagId: bid.adUnitCode,
+      sizes: bid.sizes.map(size => ({
+        w: size[0],
+        h: size[1]
+      })),
+      pageDomain: utils.getTopWindowUrl(),
+      transactionId: bid.transactionId,
+      timeout: config.getConfig('bidderTimeout')
+    };
+    const payloadString = JSON.stringify(payload);
+    return {
+      method: 'POST',
+      url: bid.params.domain + '/prebid/v1',
+      data: payloadString,
+    };
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = [];
+    if (serverResponse) {
```

Comment:
`serverResponse` is now split into two properties. `serverResponse.body` and `serverResponse.headers`.  I think you're wanting to get all the stuff off `serverResponse.body` here.